### ISSUE TITLE
steal-build watch mode error after the first event: Cannot read property 'deps' of undefined

### ISSUE THREAD

**Issue by mehiel** [stealjs/steal-tools#238]
When running in watch mode, after the first pass if I change any watched file and I get this error. 

![streal-watch-error](https://cloud.githubusercontent.com/assets/687998/7380404/47d535ca-ee04-11e4-85e0-247059d891e9.png)

**Gruntfile.js**
```javascript
'use strict';

module.exports = function (grunt) {
  require('load-grunt-tasks')(grunt, {pattern: '+(grunt-contrib-*|steal-tools|)',});

  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json'),
    "steal-build": {
      default: {
        options: {
          system: {
            config: __dirname + "/stealconfig.js",
            bundlesPath: "dist",
            main: "app/app"
          },
          buildOptions: {
            minify: true,
            bundleSteal: true,
            sourceMaps: true,
            sourceMapsContent: true,
            watch: grunt.option("watch")
          }
        }
      }
    }
  });

  grunt.registerTask('build', ['steal-build:default']);
}
```

**stealconfig.js**
```javascript
(function () {
var bowerDirectoryName = "bower_components";

steal.config({
    root: System.baseURL,
    map: {
        "jquery/jquery": "jquery",
        "can/util/util": "can/util/jquery/jquery"
    },
    paths: {
        "steal/*" : bowerDirectoryName+"/steal/*.js",
        "can/*": bowerDirectoryName+"/canjs/*.js",
        "kvsh/*": bowerDirectoryName+"/kvsh-can-extras/kvsh/*.js",

        "jquery": bowerDirectoryName+"/jquery/dist/jquery.js",
        "jquery/dom/*": bowerDirectoryName+"/jquerypp/dom/*.js",
        "jquery/event/*": bowerDirectoryName+"/jquerypp/event/*.js",
        "jquery/lang/*": bowerDirectoryName+"/jquerypp/lang/*.js",
        "jquerypp/*": bowerDirectoryName+"/jquerypp/*.js",

        "bootstrap/js/": bowerDirectoryName+"/bootstrap/js/",
        "moment-precise-range/*": bowerDirectoryName+"/moment-precise-range/*.js",
        "moment/*": bowerDirectoryName+"/moment/*.js",
        "noty/*": bowerDirectoryName+"/noty/js/noty/*.js",
        "animate/*": bowerDirectoryName+"/animate.css/*.js"
    },
    meta: {
        "jquery": { exports: "jQuery" },
        "moment/moment": { exports: "moment", format: "amd" },
        "noty/jquery.noty": { exports: "noty" }
    },
    ext: {
        ejs: "can/view/ejs/system",
        mustache: "can/view/mustache/system",
        stache: "can/view/stache/system"
    }
});

})();

System.buildConfig = {map: {"can/util/util" : "can/util/domless/domless"}};
```

**Comment by mehiel** [comment]
If this is not a silly configuration error and you can give me a clue I could start looking deeper for the reason/solution.

**Comment by justinbmeyer** [comment]
@mehiel Does it work in the browser?  What about if you run a normal build?  Thanks.

**Comment by mehiel** [comment]
Yes it works. Both in browser and normal build. If you want to repro just run this [yo generator](https://github.com/Logicea/generator-stealcanjs) and run `grunt build --watch`

**Comment by matthewp** [comment]
Thanks, I'll take a look at this, there was a bug with this module that is being fixed in `0.10.0` (released tomorrow), if you're using ES6 that could be the issue.

**Comment by mehiel** [comment]
Yes I'm all in ES6. I'll try to run that again tomorrow (it's GMT+2 here) with the latest version (or master) and I'll come back. Thank you both for the response.

**Comment by matthewp** [comment]
Hey @mehiel 0.10.0 was released, if you could update and retry let me know, this should be fixed now though.

**Comment by matthewp** [comment]
@mehiel Can you give me some more details of when it's not working?  For example I created a project using your generator and then edited a couple of files and got rebuilds, see the screenshot below.  Was it only when you edited certain files maybe?

![screen shot 2015-04-29 at 11 02 12 am](https://cloud.githubusercontent.com/assets/361671/7394076/60ef81c2-ee5f-11e4-9eb6-e82967ed9d20.png)

**Comment by mehiel** [comment]
hmmm.. interesting.. this is what I'm experiencing:

In the first screenshot I have an open file `app/app.js` (right-bottom) and I'm start building it in watch mode (left) and it all goes well and then I get the date modified and the md5 of the generated `dist/app/app.js` (right-top) for later reference.

![1-build-in-watch](https://cloud.githubusercontent.com/assets/687998/7395460/ab21b222-eea2-11e4-9381-dadd6c4634a2.png)

And now I'm going to edit the `app/app.js:31` file, remove that line and save and then I get an event on watch but nothing else happens.

![2-after-editing](https://cloud.githubusercontent.com/assets/687998/7395461/ab2a731c-eea2-11e4-9097-a95cdd4a7e9f.png)

It stays there forever and if I check the date modified and the content of the `dist/app/app.js` the date **is changed** but md5 is the same.

**Comment by mehiel** [comment]
FYI I add some environmental details that may be relevant:
**OS**: Linux 3.19.3 on Debian stretch/sid
**filesystem**: ext4
**nodejs**: v0.10.29
**grunt**: v0.1.13

_If you want anything else just let me know,_
Thanks!

**Comment by matthewp** [comment]
The event in the console is the indication that a rebuild has occurred.  Maybe that needs to be clarified in the docs. There is a timestamp and the name of the module that triggered a rebuild.  Nothing else should happen (we don't have the verbose output with watch mode).  When you reload the page do you not see your change?

**Comment by mehiel** [comment]
Yes I got that. It seems that it's start building but hangs. As you can see from the md5 in the screens above, the content is not changing (and as a consequence, no if I reload the page I don't see my change). And the thing is that if I do more changes I get no events after the first.

**Comment by mehiel** [comment]
I'll try to run that under `strace` to see if I get more info for debugging.

**Comment by matthewp** [comment]
Maybe try with quiet: false and verbose: true  watch mode is intentially quiet.

**Comment by mehiel** [comment]
The error described in this issue is fixed by 0.10.0
